Bump Cloud Trace library to v1.8.0

damemi opened the pull request:
Fixes https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/208

This bumps the cloud trace client to v1.8.0, which offers default batch retry on failed writes.

Currently the trace client [retries failed calls](https://github.com/googleapis/google-cloud-go/blob/trace/v1.2.0/trace/apiv2/trace_client.go#L59) for `CreateSpans` but not `BatchWriteSpans`.

dashpole commented:
Should we open an upstream issue asking for the default to be changed?  Or ask why the default is set the way it is?

If we can avoid adding a new config surface, that seems best.  It also would seem odd for us to have a different default from the upstream library.

damemi commented:
Upstream issue raised at https://github.com/googleapis/google-cloud-go/issues/7184

damemi commented:
update: we implemented this as a default setting in the upstream trace client instead. but, I think we can still use the unit tests I added here to make sure attempts are retried. once the upstream generated clients land, I'll use this PR to bump the dependency and add these tests. Holding until then
